File: day_08/python/part2.py
import re

strings = open('../input.txt').read().splitlines()

strlen_original = 0
strlen_encoded = 0

for string in strings:
    strlen_original += len(string)
    # encoded = eval(string)
    encoded = '"' + re.sub(
        r'(["\\])',
        r'\\\1',
        string
    ) + '"'
    strlen_encoded += len(encoded)

strlen_diff = strlen_encoded - strlen_original

print('Diff: {}'.format(strlen_diff))

File: day_08/php/part2.php
<?php

$lines = file('../input.txt');

$strlen_original = 0;
$strlen_encoded = 0;

foreach ($lines as $line) {
    $original = trim($line);
    $encoded = '"' . addslashes($original) . '"';

    // var_dump(sprintf('{%1$s} => {%2$s}', $original, $encoded));

    $strlen_original += strlen($original);
    $strlen_encoded += strlen($encoded);
}

$strlen_diff = $strlen_encoded - $strlen_original;

echo sprintf('%1$s', $strlen_diff) . PHP_EOL;
